tools/c7n_azure/c7n_azure/resources/sqldatabase.py

```diff
@@ -238,3 +271,174 @@ def get_retention_from_backup_policy(self, retention_policy):
         if actual_duration_units.iso8601_symbol != self.retention_period_units.iso8601_symbol:
             return None
         return actual_duration
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BackupRetentionPolicyAction(AzureBaseAction):
+
+    def __init__(self, operations_property, *args, **kwargs):
+        super(BackupRetentionPolicyAction, self).__init__(*args, **kwargs)
+        self.operations_property = operations_property
+
+    def _process_resource(self, database):
+        client = self.manager.get_client()
```

Comment:
Please move `get_client()` call to the `_prepare_processing()`, client can be shared